File: src/lessons/bat/string1/StringBitsEntity.java
package lessons.bat.string1;

import static plm.core.ValueSerializer.*;

import plm.universe.bat.BatEntity;

public class StringBitsEntity extends BatEntity {

  public void run()
  {
    int count = getTestCount();
    for (int i = 0; i < count; i++) {
      Object[] param = (Object[])deserialize(getTest(i));
      setTestResult(i, serialize(stringBits((String)param[0])));
    }
  }

  /* BEGIN TEMPLATE */
  String stringBits(String str)
  {
    /* BEGIN SOLUTION */
    String result = "";
    // Note: the loop increments i by 2
    for (int i = 0; i < str.length(); i += 2) {
      result = result + str.substring(i, i + 1);
      // Alternately could use str.charAt(i)
    }
    return result;
    /* END SOLUTION */
  }
  /* END TEMPLATE */
}

File: src/lessons/bat/string1/StringBitsEntity.py
from RemoteBat import *
def run():
    count = getTestCount()
    for i in range(count):
        param = deserialize(getTest(i))
        setTestResult(i, serialize(stringBits(param[0])))

# BEGIN TEMPLATE
def stringBits(str):
# BEGIN SOLUTION
  res = ''
  for i in range(0,len(str),2):
    res += str[i:i+1]
  return res
# END SOLUTION
# END TEMPLATE
